cmd/gb-list/main.go

```diff
@@ -47,9 +56,27 @@ func main() {
 	if err != nil {
 		gb.Fatalf("unable to resolve: %v", err)
 	}
-	for _, pkg := range pkgs {
-		if err := tmpl.Execute(os.Stdout, pkg); err != nil {
-			gb.Fatalf("unable to execute template: %v", err)
+
+	if jsonOutput {
+		views := make([]*PackageView, 0, len(pkgs))
+		for _, pkg := range pkgs {
+			views = append(views, NewPackageView(pkg))
+		}
+		encoded, err := json.MarshalIndent(views, " ", "  ")
```

Comment:
Does it need to be indented ? I'm assuming the json form is only for
machines to consume, and they don't care about pretty printing.

How about this, use json.NewEncoder and don't indent. People that want it
indented can pipe it through jsonpp. We can always make this better later.

On Wed, May 13, 2015 at 10:01 PM, Mario Kostelac <notifications@github.com>
wrote:

> In cmd/gb-list/main.go
> <https://github.com/constabulary/gb/pull/77#discussion_r30223723>:
>
> > @@ -47,9 +56,27 @@ func main() {
> >  	if err != nil {
> >  		gb.Fatalf("unable to resolve: %v", err)
> >  	}
> > -	for _, pkg := range pkgs {
> > -		if err := tmpl.Execute(os.Stdout, pkg); err != nil {
> > -			gb.Fatalf("unable to execute template: %v", err)
> > +
> > +	if jsonOutput {
> > +		views := make([]*PackageView, 0, len(pkgs))
> > +		for _, pkg := range pkgs {
> > +			views = append(views, NewPackageView(pkg))
> > +		}
> > +		encoded, err := json.MarshalIndent(views, " ", "  ")
>
> I will try again. I was not able to find a way to write prettified JSON
> with pure json encoder.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/77/files#r30223723>.
>